internal/clients/ec2/ec2_client.go

```diff
@@ -24,7 +24,7 @@ type Client struct {
 	log     zerolog.Logger
 }
 
-func NewEC2Client(ctx context.Context) *Client {
+func NewEC2Client(ctx context.Context, region ...string) *Client {
```

Comment:
isn't this general args array?
I believe `params ...string` would describe the param bit better.
And then ofc somewhere `region = params[0]`

In case you wanted to add only region, I believe we should do `region string` (not an array).
Or am I missing something?

Reply:
it is a variadic function, so we won't need to change all the calls in the jobs. 
in the next step (adding the region to the jobs) we can remove it and use a regular region parameter. 
I didn't think that using the optional struct https://bojanz.github.io/optional-parameters-go/ is good practice here